src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const tempArr = date.split(fromFormat[3]);
+  const resultArr = [];
+
+  for (let i = 0; i <= 2; i++) {
```

Comment:
The loop assumes that there are always three components in the date format (e.g., day, month, year). This might not hold true for all date formats. Consider making the loop adaptable to different numbers of components.